packages/builder/src/builtin-tasks/compile.ts

```diff
@@ -0,0 +1,72 @@
+import { task } from "../internal/core/config/config-env";
+import { networks } from "../../sample-project/algob.config";
+import { TASK_COMPILE } from "./task-names";
+import YAML from "yaml";
+const algosdk = require('algosdk');
+const fs = require('fs');
+const path = require('path');
+const murmurhash = require('murmurhash');
+
+/* Credentials */
+const token = networks.localhost.token;
+const host = networks.localhost.host;
+const port = networks.localhost.port;
+
+let algodclient = new algosdk.Algodv2(token, host, port);
+
+export default function (): void {
+  task(TASK_COMPILE, "Compilation task")
+    .setAction(
+      async () => {
+        console.log("compiling......");
+        /*File paths for assets/ and artifacts/cache/ */
+        var srcPath = path.join(__dirname, '../../sample-project/assets/');
```

Comment:
Nope.
In short: you shouldn't change the default project, only the one that is in current directory. i.e. we check that we're in a project.

This should be current directory, like this: `path.join(__dirname, 'assets/')`.
The command is prevented to run out of a project folder.
So first of all you have to create a project with `init` and then you'll be able to use the command `compile` in that directory.
The config variables should be available from `algobEnv` variable which you didn't specify in the `setAction` async function input (line 20).

Reply:
Also you may want to create that directory so that it will exist. It belongs to you, it's not part of the sample project (why would sample project ship with compiled code? it shouldn't, compile task should create a directory for it.)
(think about command `mkdir -p dir/dir/dir` -- it will create all intermediate directories.